types/index.d.ts

```diff
@@ -0,0 +1,9 @@
+// It’s important to preserve this ignore statement. This maeks sure it works
+// both with and without node types.
+// @ts-ignore
```

Comment:
ts-expect-error maybe?

Reply:
It **may not** yield a type error if `@types/node` is loaded, but it **must** yield a type error otherwise.

This means `ts-ignore` is the only option. It’s hacky, because there is no proper solution.